great_expectations/data_context/types/base.py

```diff
@@ -939,7 +939,7 @@ class Meta:
         required=False,
         allow_none=True,
     )
-    id = fields.String(
+    id_ = fields.String(
```

Comment:
Using `id` was causing issues with `instantiate_class_from_config`. We may wish to refactor to use `id` in the future.

Reply:
If it's simpler this way, I'm totally fine with it. Obviously we'd prefer `id` but I don't think it's worth spinning our wheels over it.